api/app.js

```diff
@@ -63,7 +63,33 @@ module.exports = function(configFilePath) {
         res.status(HttpStatus.BAD_REQUEST).json(errorResponse);
       });
         
+  });
+
+  var gracefulShutdown = function(signal) {
+    logger.info('Received ' + signal + ' signal, shutting down gracefully...');
+    server.shutdown(function() {
+      logger.info('Everything is cleanly shutdown');
+      process.exit();
+    })
+    
+    setTimeout(function(){
+      server.forceShutdown(function(){
+        logger.info('Could not close connections in time, forcefully shutting down');
+        process.exit();
+      })
+    }, 10*1000);
```

Comment:
Make a separate PR to clean this up with a constant.
Realize that when this is integrated, drain will issue SIGUSR2 until some time period and invoke a SIGKILL.